### DISCUSSION TITLE
How to combine two different plots using `seaborn.objects`?

### DISCUSSION THREAD

**Discussion by paw-lu** [mwaskom/seaborn#3335] [Q&A] [answered]
When using the newish `seaborns.objects` inferface—which is great by the way—is there a cannonical way to combine two plots? Best way I can think of now is to create an `Axes` and plot both of them on it.

If this is the best way, what's the easiest way to assign a label to each line here so that there's a legend that distinguishes each line?

```python
import numpy as np
import seaborn.objects as so

_, ax = plt.subplots(figsize=(20, 10))
rng = np.random.default_rng()
data1 = rng.weibull(1, size=5_000)
data2 = rng.weibull(1.3, size=5_000)
(
    so.Plot(data1)
    .add(so.Line(color="C0",), so.KDE(bw_adjust=2))
    .on(ax)
    .plot()
)
(
    so.Plot(data2)
    .add(so.Line(color="C1"), so.KDE(bw_adjust=2))
    .on(ax)
    .plot()
)
```

Thanks!

**Comment by paw-lu** [comment]
Quick aside: I really like altair's API on combining plots, where if you have `chart1` and `chart2`, you can combine them via `chart1 + chart2`, have them side by side via `chart1 | chart2`, stacked using `chart1 & chart2` etc.

**Comment by mwaskom** [comment, marked as answer]
Is what you're looking for here a layered plot?

```python
(
    so.Plot()
    .add(so.Line(color="C0"), so.KDE(bw_adjust=2), x=data1)
    .add(so.Line(color="C1"), so.KDE(bw_adjust=2), x=data2)
)
```
<img width=400 src=https://user-images.githubusercontent.com/315810/233754895-69a0373c-5ced-4ba9-89f1-b192dffd50a1.png />

**Reply by paw-lu** [reply]
> Is what you're looking for here a layered plot?

Yep, this works perfectly. Embarassingly I didn't realize that you could initialize `so.Plot()` with no args, then pass `x` to `add` instead with the data.

**Reply by paw-lu** [reply]
Thanks a lot!

**Reply by mwaskom** [reply]
Cool, also note that the "layer legend" aspect is still WIP: #3046

**Comment by tburnett** [comment]
Please add a comment about this use case to the Plot documentation. And perhaps to the tutorial as well.  I had an issue with overplotting a line on a histogram. With only the x source specified in the constructor, it was not possible to then set x and y in the add.
But overlayed plots without  legends are almost useless for presentations or publications.
